core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -452,10 +458,23 @@ private void handleRequest(ChannelHandlerContext ctx, DecodedHttpRequest req) th
     private static HttpResponse serve0(HttpRequest req,
                                        ServiceConfig serviceCfg,
                                        HttpService service,
-                                       DefaultServiceRequestContext reqCtx) {
+                                       DefaultServiceRequestContext reqCtx,
+                                       ServerMetrics serverMetrics,
+                                       boolean isHttp1WebSocket) {
         try (SafeCloseable ignored = reqCtx.push()) {
             HttpResponse serviceResponse;
             try {
+                if (isHttp1WebSocket) {
+                    serverMetrics.increaseActiveHttp1WebSocketRequests();
+                    serverMetrics.decreasePendingHttp1Requests();
+                } else if (reqCtx.sessionProtocol().isExplicitHttp1()) {
+                    serverMetrics.increaseActiveHttp1Requests();
+                    serverMetrics.decreasePendingHttp1Requests();
+                } else if (reqCtx.sessionProtocol().isExplicitHttp2()) {
+                    serverMetrics.increaseActiveHttp2Requests();
+                    serverMetrics.decreasePendingHttp2Requests();
+                }
```

Comment:
Sorry, I think the current approach looks a bit unappealing. I've changed my mind. 😓 How do you feel about incorporating this logic into the decoded request instead?
```java
class AggregatingDecodedHttpRequest {
    AggregatingDecodedHttpRequest(...) {
        ...
        if (routingCtx.sessionProtocol().isMultiplex()) {
            metrics.increasePendingHttp2Requests();
        } else {
            metrics.increasePendingHttp1Requests();
        }
    }
    ...
    private void completeAggregationFuture() {
        if (aggregationFuture.complete(null)) {
            if (routingCtx.sessionProtocol().isMultiplex()) {
                metrics.decreasePendingHttp2Requests();
                metrics.increaseActiveHttp2Requests();
            } else {
                metrics.decreasePendingHttp1Requests();
                metrics.decreaseActiveHttp1Requests();
            }
        }
    }
}

class StreamingDecodedHttpRequest {
    StreamingDecodedHttpRequest(...) {
        ...
        if (http1WebSocket) {
            metrics.increaseActiveHttp1WebSocketRequests();
        } else if (routingCtx.sessionProtocol().isMultiplex()) {
            metrics.increaseActiveHttp2Requests();
        } else {
            metrics.increaseActiveHttp1Requests();
        }
    }

}

class EmptyContentDecodedHttpRequest {...}
```

Reply:
I'm so sorry to take it so long to apply the review!
i've been trying to write more test cases related to AggregatingDecodedHttpRequest, StreamingDecodedHttpRequest and StreamingDecodedHttpRequest.

but it's a little bit hard and taking some time..
i will do my best to apply the review asap!